provider/consul/consul_catalog.go

```diff
@@ -202,22 +230,38 @@ func (p *CatalogProvider) getBackendName(node *api.ServiceEntry, index int) stri
 }
 
 func (p *CatalogProvider) getAttribute(name string, tags []string, defaultValue string) string {
+	return p.getTag(p.getPrefixedName(name), tags, defaultValue)
+}
+
+func (p *CatalogProvider) hasTag(name string, tags []string) bool {
+	tag := p.getTag(name, tags, "=!=") // Very-very unlikely that a Consul tag would ever start with '=!='
```

Comment:
This is my internal dummy "default" value (because `getTag` function want's a default) which is unlikely to be seen in the real world (who would tag their service with `=!=`... :) )
The logic is this: if `getTag(...)` returns `=!=`, then there is no `name` tag in existence, otherwise we would get a different value.